domain/src/test/java/org/oppia/android/domain/platformparameter/syncup/PlatformParameterSyncUpWorkerTest.kt

```diff
@@ -322,6 +326,89 @@ class PlatformParameterSyncUpWorkerTest {
       .containsEntry(TEST_INTEGER_PARAM_NAME, defaultTestIntegerParameter)
   }
 
+  @Test
+  fun testSyncUpWorker_getFeatureFlags_addSyncStatusFlags_verifyCorrectStatusReturned() {
+    // Set up versionName to get correct network response from mock platform parameter service.
+    setUpApplicationForContext(MockPlatformParameterService.appVersionForCorrectResponse)
+
+    // Empty the Platform Parameter Database to simulate the execution of first SyncUp Work request.
+    platformParameterController.updatePlatformParameterDatabase(listOf())
+
+    val workManager = WorkManager.getInstance(context)
+
+    val inputData = Data.Builder().putString(
+      PlatformParameterSyncUpWorker.WORKER_TYPE_KEY,
+      PlatformParameterSyncUpWorker.PLATFORM_PARAMETER_WORKER
+    ).build()
+
+    val request: OneTimeWorkRequest = OneTimeWorkRequestBuilder<PlatformParameterSyncUpWorker>()
+      .setInputData(inputData)
+      .build()
+
+    // Enqueue the Work Request to fetch and cache the Platform Parameters from Remote Service.
+    workManager.enqueue(request)
+    testCoroutineDispatchers.runCurrent()
+
+    val workInfo = workManager.getWorkInfoById(request.id)
+    assertThat(workInfo.get().state).isEqualTo(WorkInfo.State.SUCCEEDED)
+
+    // Retrieve the previously cached Platform Parameters from Cache Store.
+    monitorFactory.ensureDataProviderExecutes(platformParameterController.getParameterDatabase())
+
+    // Values retrieved from Cache store will be sent to Platform Parameter Singleton by the
+    // Controller in the form of a Map, therefore verify the retrieved values from that Map.
+    val platformParameterMap = platformParameterSingleton.getPlatformParameterMap()
+
+    // Previous String Platform Parameter is still same in the Database.
+    assertThat(platformParameterMap)
+      .containsEntry(TEST_STRING_PARAM_NAME, expectedTestStringParameter)
+
+    // Is synced boolean of the platform parameter is true
+    assertThat(platformParameterMap[TEST_STRING_PARAM_NAME]!!.isSynced).isEqualTo(true)
```

Comment:
```suggestion
    assertThat(platformParameterMap[TEST_STRING_PARAM_NAME]?.isSynced).isTrue()
```

Reply:
Same for this. Booleans are no longer being used so this change cannot be made.